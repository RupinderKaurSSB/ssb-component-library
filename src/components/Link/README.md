Link
========

> A page layout component to wrap around lots of content 
### Usage


#### HTML

```html
<a class="ssb-link" href=" " target="" rel="">
    <span class="link-text">A regular link</span>
</a>
```

#### React

```jsx harmony
<Link href=" ">A regular link</Link>
```

Available props:

| Name       | Type           | Description  |
| ---------- | ------------- | ----- |
| children   | Required node | Content within the wrapper |
| href | Required string|  |
| icon | Optional node |  |
| isExternal | Optional boolean |  |
| linkType | Optional oneOf | One of 'regular', 'profiled' or 'header'  |
| negative | Optional boolean | Styling for dark theme |
