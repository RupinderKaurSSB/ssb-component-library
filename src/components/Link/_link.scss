$default-font-size: 16px;
$profiled-font-size: 18px;
$header-font-size: 20px;
$icon-padding: 8px;

.ssb-link {
  @include open-sans;
  align-items: center;
  background-image: linear-gradient(120deg, $ssb-green-4 0%, $ssb-green-4 100%);
  background-position: 0 100%;
  background-repeat: no-repeat;
  background-size: 100% 0;
  border-bottom: 1px solid $ssb-green-4;
  color: $ssb-green-4;
  cursor: pointer;
  display: inline-flex;
  font-size: $default-font-size;
  line-height: 1.3;
  padding: 2px 2px 1px;
  position: relative;
  text-decoration: none;
  transition: background-size .2s ease-in, color .1s;

  &.header {
    @include roboto;
    font-size: $header-font-size;
    font-weight: bold;

    .link-text {
      color: $ssb-dark-6;
    }
  }

  &:hover, &:focus {
    background-size: 100% 100%;
    color: $ssb-white;
    outline: 0;
    text-decoration: none;

    .link-text {
      color: $ssb-white;
    }
  }

  &.negative {
    background-image: linear-gradient(120deg, $ssb-green-2 0%, $ssb-green-2 100%);
    border-bottom: 1px solid $ssb-green-2;
    color: $ssb-green-2;

    .link-text {
      color: $ssb-green-2;
    }

    &.header {

      .link-text {
        color: $ssb-white;
      }
    }

    &.with-icon {
      padding: 2px 2px 0;
      border-bottom: none;

      .link-text {
        color: $ssb-white;
      }

      &:hover, &:focus {
        color: $ssb-dark-5;

        .link-text {
          color: $ssb-dark-5;
        }
      }

    }

    &:hover, &:focus {
      color: $ssb-dark-5;
      outline: 0;

      .link-text {
        color: $ssb-dark-5;
      }
    }
  }

  &.profiled {
    font-size: $profiled-font-size;
    font-weight: bold;
  }

  &.with-icon {
    padding: 2px 2px 0;
    border-bottom: none;

    .link-text {
      color: $ssb-dark-6;
    }

    &:hover, &:focus {
      color: $ssb-white;

      .link-text {
        color: $ssb-white;
      }
    }
  }

  .link-text {
    color: $ssb-green-4;
    transition: color .1s;

    &:hover, &:focus {
      color: $ssb-white;
      outline: 0;
    }
  }
}
