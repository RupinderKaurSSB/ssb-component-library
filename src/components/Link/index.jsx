import React from 'react';
import PropTypes from 'prop-types';

const Link = ({
	ariaLabel, children, className, href, icon, isExternal, linkType, negative, tabIndex,
}) => {
	const classNames = `ssb-link${linkType ? ` ${linkType}` : ''}${negative ? ' negative' : ''}${icon ? ' with-icon' : ''}${className ? ` ${className}` : ''}`;

	return (
		<a
			className={classNames}
			href={href}
			target={isExternal ? '_blank' : ''}
			rel={isExternal ? 'noopener noreferrer' : ''}
			tabIndex={tabIndex}
			aria-label={ariaLabel}
		>{icon && <div className="icon-wrapper">{icon}</div>}
			{children && <span className="link-text">{children}</span>}
		</a>
	);
};

Link.defaultProps = {
	className: '',
	isExternal: false,
	negative: false,
};

Link.propTypes = {
	ariaLabel: PropTypes.string,
	children: PropTypes.node,
	className: PropTypes.string,
	href: PropTypes.string.isRequired,
	icon: PropTypes.node,
	isExternal: PropTypes.bool,
	linkType: PropTypes.oneOf([
		'regular',
		'profiled',
		'header',
	]),
	negative: PropTypes.bool,
	tabIndex: PropTypes.number,
};

export default Link;
