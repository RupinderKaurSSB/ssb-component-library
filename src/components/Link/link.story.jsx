import React from 'react';
import { storiesOf } from '@storybook/react';
import centered from '@storybook/addon-centered/react';
import { ArrowRight } from 'react-feather';
import Link from './index';

storiesOf('Link', module).addDecorator(centered)
	.add('Default', () => (
		<div>
			<p><Link href=" ">A regular link</Link></p>
			<Link href=" " icon={<ArrowRight size={20} />}>A regular link with icon</Link>
		</div>
	))
	.add('Profiled', () => (
		<div>
			<p><Link href=" " linkType="profiled">A profiled link</Link></p>
			<Link href=" " linkType="profiled" icon={<ArrowRight />}>A profiled link with icon</Link>
		</div>
	))
	.add('Header', () => (
		<div>
			<p><Link href=" " linkType="header">A header link</Link></p>
		</div>
	));
