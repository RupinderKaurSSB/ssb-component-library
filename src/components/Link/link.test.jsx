import React from 'react';
import {shallow} from 'enzyme';
import Link from './index';
import { ArrowRight } from 'react-feather'

describe('Link component', () => {
	test('Matches the snapshot', () => {
		const wrapper = shallow(<Link href="">A regular link</Link>);
		expect(wrapper).toMatchSnapshot();
	});
	test('Toggles classNames correctly', () => {
		const wrapper = shallow(<Link className="mt-3" isExternal href="" negative linkType="profiled">Profile link</Link>);
		expect(wrapper.hasClass('mt-3')).toEqual(true);
		expect(wrapper.hasClass('negative')).toEqual(true);
		expect(wrapper.hasClass('profiled')).toEqual(true);
		expect(wrapper.props().rel).toEqual('noopener noreferrer');
	});
	test('render icon', () => {
		const wrapper = shallow(<Link href=" " linkType="profiled" icon={<ArrowRight size={20} />}>A profiled link with icon</Link>);
		expect(wrapper.hasClass('with-icon')).toEqual(true);
		expect(wrapper.containsMatchingElement(<ArrowRight size={20} color="currentColor" />)).toEqual(true);
	});

});
