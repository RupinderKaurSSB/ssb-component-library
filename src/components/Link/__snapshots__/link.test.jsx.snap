// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component Matches the snapshot 1`] = `
<a
  className="ssb-link"
  href=""
  rel=""
  target=""
>
  <span
    className="link-text"
  >
    A regular link
  </span>
</a>
`;
